src/front/wgsl/lower/mod.rs

```diff
@@ -510,7 +510,13 @@ impl<'source, 'temp, 'out> ExpressionContext<'source, 'temp, 'out> {
         left: &mut Handle<crate::Expression>,
         right: &mut Handle<crate::Expression>,
     ) -> Result<(), Error<'source>> {
-        if op != crate::BinaryOperator::Multiply {
+        if matches!(
+            op,
+            crate::BinaryOperator::Add
+                | crate::BinaryOperator::Subtract
+                | crate::BinaryOperator::Divide
+                | crate::BinaryOperator::Modulo
```

Comment:
Looking at the table in the spec, `Multiply` should also work (not sure why it was previously the only operation ignored).

Note that the comment on the function also needs to be updated.

Reply:
@teoxoy `Multiply` is excluded here as (AFAIK) the backends are expected to handle vec*scalar operations.

We could insert splats here and it would still work, but would increase IR and output size needlessly, as in e.g.:

```diff
-    gl_Position = (((vec4(1.0) * _e4) * _e8) * _e12);
+    gl_Position = (((vec4(1.0) * vec4(_e4)) * vec4(_e9)) * vec4(_e14));
```

The [typifier](https://github.com/gfx-rs/naga/blob/master/src/proc/typifier.rs#L564-L612) handles vec*scalar (as well as matrix cases) for type checking.

I've added d848fdd6fce1b3062ca22122504ec949aec4cb92 to expand the function comment with this information, and also adding a reference to the specification.

Let me know what you think!

Reply:
Ah, I see - thanks for the clarification!